### PR TITLE
docs: check for dropped urls in PRs

### PR DESCRIPTION
**Pull request by MikhailKardash**
## Description

`make docs publish` requires that all previously-published URLs have a redirect, even if that redirect is to something like `_index.html`.  This PR makes sure that this condition will be checked in `redirects.py check` and consequently `make docs check` to catch this error before `redirects.py publish` is attempted.

Also add some redirects for some dropped previously-published urls.

## Test Plan

1. Navigate to `determined/docs`
2. Run `python redirects.py check`. There should be no errors.
3. Delete a file, such as `docs/integrations/prometheus/_index.rst`
4. Run `python redirects.py check`. There should be errors for broken redirects AND errors for dropped published urls. 

## Checklist

- [X] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
MLG-1100

### CODE REVIEW

File: docs/redirects.py [hunk 1]
```diff
@@ -290,12 +290,25 @@ def all_urls_from_files():
         exit(1)
 
     if sys.argv[1] == "check":
+        #check for broken links
         errors = check_links(links)
         if errors:
             print("check failed, errors detected!", file=sys.stderr)
             for e in errors:
                 print(e, file=sys.stderr)
             exit(1)
```
Comment (rb-determined-ai) [inline docs/redirects.py:299]: is there a way to return all errors at once, instead of exiting before checking for dropped urls?

Comment (MikhailKardash) [reply docs/redirects.py]: Good point, I switched it to have a return_code variable.

**Review by tara-det-ai — APPROVED** [review]
LGTM

**Review by rb-determined-ai — APPROVED** [review]
excellent, this is a nice improvement to the ci checks

**Pull request merged**